src/python/pybind11/src/pyxrt.cpp

```diff
@@ -105,7 +105,11 @@ py::class_<xrt::run>(m, "run")
 py::class_<xrt::kernel>(m, "kernel")
     .def(py::init([](xrt::device d, const py::array_t<unsigned char> u, const std::string & n, bool e){
         return new xrt::kernel(d, (const unsigned char*) u.request().ptr, n, e);
```

Comment:
I am not claiming to understand this file, but where is the corresponding delete?

Reply:
with this return style, Python takes ownership of the object and calls delete() when the Python object's reference counter goes to zero.